devdoc/rc_ptr.md

```diff
@@ -0,0 +1,34 @@
+# `rc_ptr` requirements
+
+## Overview
+
+`rc_ptr` is a module that encapsulates reference-counted pointers.
+
+## Exposed API
+```c
+typedef void (*RC_PTR_FREE_FUNC)(void* context);
+
+typedef struct RC_PTR_TAG
+{
+    const void* ptr;
+    RC_PTR_FREE_FUNC free_func;
+} RC_PTR;
```

Comment:
Why is this struct part of the "exposed API"? Can there be some accessor to the "ptr" API? and then the whole structure can go away? 

It exposes "free_func" which I don't think is ever intended to be accessed (called) by the user.

Reply:
That's how `rc_string` does it. 

`free_func` will be removed from the exposed struct in the final implementation by use of an internal struct in the `.c` file. 

Just wanted some quick feedback on the API so didn't implement it that way yet.